coder_sniffer/Drupal/Sniffs/Commenting/DeprecatedSniff.php

```diff
@@ -75,20 +74,72 @@ public function process(File $phpcsFile, $stackPtr)
+            // The full text does not match the standard. Try to find fixes by
+            // testing with a relaxed set of criteria, based on common
+            // formatting variations. This is designed for Core fixes.
+            $error = "The text '@deprecated %s' does not match the standard format: ".$standardFormat;
+            // All of the standard text should be on the first comment line, so
+            // try to match with common formatting errors to allow an automatic
+            // fix. If not possible then report a normal error.
+            $matchesFix = [];
+            $fix        = null;
+            if (count($textItems) > 0) {
+                // Get just the first line of the text.
+                $key   = array_keys($textItems)[0];
+                $text1 = $textItems[$key];
+                preg_match('/^(.*)(as of|in) (drupal|)( |:|)+([\d\.\-xdev\?]+)(,| |. |)(.*)(removed|removal)([ |from|before|in|the]*) (drupal|)( |:|)([\d\-\.xdev]+)( |,|$)+(?:release|)(?:[\.,])*(.*)$/i', $text1, $matchesFix);
+
+                if (count($matchesFix) >= 12) {
+                    // This problem is a drupal core file and is fixable.
+                    if (empty($matchesFix[1]) === false) {
+                        // Verify that its acceptable to remove the text in [1].
+                        echo('>>> File: '.$phpcsFile->path.' line '.$tokens[($stackPtr)]['line'].PHP_EOL);
+                        echo('>>> First line: '.$text1.PHP_EOL);
+                        echo('>>> Would remove: '.$matchesFix[1].PHP_EOL);
+                    }
+
+                    $ver1 = str_Replace(['x-dev', 'x'], ['0', '0'], trim($matchesFix[5], '.'));
+                    $ver2 = str_Replace(['x-dev', 'x'], ['0', '0'], trim($matchesFix[12], '.'));
+                    // If the version is short, add enough '.0' to correct it.
+                    while (substr_count($ver1, '.') < 2) {
+                        $ver1 .= '.0';
+                    }
+
+                    while (substr_count($ver2, '.') < 2) {
+                        $ver2 .= '.0';
+                    }
+
+                    $correctedText = trim('in drupal:'.$ver1.' and is removed from drupal:'.$ver2.'. '.trim($matchesFix[14]));
+                    // If $correctedText is longer than 65 this will make the whole
+                    // line exceed 80. It is probably longer than 80 before the fix.
+                    if (strlen($correctedText) > 65) {
+                        echo('>>> WARNING: File '.$phpcsFile->path.' line '.$tokens[($stackPtr)]['line'].PHP_EOL);
+                        echo('>>> WARNING: $correctedText = '.$correctedText.PHP_EOL);
+                        echo('>>> WARNING: length of $correctedText = '.strlen($correctedText).' (so line will exceed 80)'.PHP_EOL);
+                    }
+
+                    $fix = $phpcsFile->addFixableError($error, $key, 'IncorrectTextLayoutFixable', [$fullText]);
```

Comment:
we should not have different error codes for fixable errors. Just use 1 code.

Reply:
OK